app/components/ToggleDisplay.js

```diff
@@ -80,28 +79,17 @@ export default class ToggleDisplay extends Component {
 
   runHelpers = () => {
     if (this.state.helpers) {
-      let shouldHide = false;
-      let obj = {};
-
       chrome.storage.local.get('display', result => {
-        this.state.helpers.map(helper => {
+        this.state.helpers.forEach(helper => {
           if (helper.type === 'hide') {
             if (!{}.hasOwnProperty.call(result.display, helper.relation)) {
-              return helper;
+              return;
             }
             const relation = result.display[helper.relation].visible;
-            shouldHide = this.state.visible + relation;
-            obj = helper;
-            obj.display = true;
+            helper.display = (this.state.visible + relation) !== 0;
+            this.sendHelperToDOM(helper);
           }
```

Comment:
Yeah we're good here. Your refactoring hasn't actually changed anything for cases where the [helpers](https://github.com/ChessCom/browser-extension/blob/fb5abf897e5ddacb1c8742e7d7d8bd7e5c144c9b/app/components/Options/Config.js#L282-L303) length is just 1, which is all the current cases. We're good here.